node-agent/workload_manager/src/workload_manager/workload_listener/workload_listener.rs

```diff
@@ -0,0 +1,14 @@
+use proto::agent::{Instance, InstanceStatus};
+use std::sync::mpsc::Sender;
+use tonic::Status;
+
+pub trait WorkloadListener {
+    /// Launch a thread that will listen to a workload and send continously an InstanceStatus
+    ///
+    /// # Arguments
+    ///
+    /// * `id` - A String that is used by the workload's engine to identify it (docker containers' id for instance)
+    /// * `instance` - An Instance struct, given by the scheduler,
+    /// * `sender` - A sender given by the WorkloadManager, whose receiver is given to the scheduler
+    fn new(id: String, instance: Instance, sender: Sender<Result<InstanceStatus, Status>>) -> Self;
```

Comment:
You don't need a trait for a constructor. A trait represents a behaviour, creating an instance is not a behaviour. Just use `ContainerListener::new` directly.

Reply:
would something like this be better ? https://github.com/do3-2021/kudo/blob/tests-node-agent/node-agent/workload_manager/src/workload_manager/workload_listener/mod.rs

Reply:
That file is fine. It shows that you don't need that trait at all.